src/commands/deployments/create.ts

```diff
@@ -17,19 +17,18 @@ export const createDeployment = async (
 ): Promise<Deployment> => {
 	const client = getDeploymentApiClient(args.token);
 	try {
-		const lastDeployment = await client.getLatestDeploymentDeprecated({
-			appId: args.appId,
-		});
-
+		let lastDeployment: Deployment | undefined;
 		if (
-			lastDeployment === undefined &&
-			(args.buildId === undefined ||
-				args.roomsPerProcess === undefined ||
-				args.planName === undefined ||
-				args.transportType === undefined ||
-				args.containerPort === undefined)
+			args.buildId === undefined ||
+			args.roomsPerProcess === undefined ||
+			args.planName === undefined ||
+			args.transportType === undefined ||
+			args.containerPort === undefined
 		) {
-			throw new Error("All args must be present for the initial deployment");
+			console.log("Some args missing, copying values from the last deployment");
+			lastDeployment = await client.getLatestDeploymentDeprecated({
```

Comment:
couldn't `lastDeployment` still be undefined here?

Reply:
it won't be undefined, but it will throw an error if no previous deployment exists. it won't be the best error message for the user, but that is an expected error case, and I added the `console.log` which will provide good insight in case users hit that error

Reply:
previously the bug was that we were calling `client.getLatestDeployment` no matter what, now we're only calling it if you're missing some args. as long as you pass in all the required args, we won't make that API call after this change

Reply:
Ah got it, `lastDeployment` is never undefined, the endpoint throws. Should we catch and display a better error? Can we also remove `lastDeployment!` below? It shouldn't be needed

Reply:
the `!` is still needed because `lastDeployment` would be undefined in the case where we never enter the `if` case

Reply:
Ahh yes, this code is a lil awkward but good enough